Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -417,44 +393,45 @@ void UPSWorldSubsystem::ResetSaveGameData()
 	const FString& SlotName = UPSSaveGameData::GetSaveSlotName();
 	const int32 UserIndex = UPSSaveGameData::GetSaveSlotIndex();
 
-	UGameplayUtilsLibrary::ResetSaveGameData(SaveGameDataInternal, SlotName, UserIndex);
-
-	// Re-load a new save game object. Load game from save creates a save file if there is no such
-	LoadGameFromSave();
-	SetCurrentRowByTag(PSCurrentSpotComponentInternal->GetMeshChecked().GetPlayerTag());
-	RefreshProgressionUIElements();
-}
+	SaveGameDataInternal = Cast<UPSSaveGameData>(UGameplayUtilsLibrary::ResetSaveGameData(SaveGameDataInternal, SlotName, UserIndex));
 
-// Unlocks all levels of the Progression System
-void UPSWorldSubsystem::UnlockAllLevels()
-{
-	if (!ensureMsgf(SaveGameDataInternal, TEXT("ASSERT: [%i] %s:\n'SaveGameDataInternal' is not valid!"), __LINE__, *FString(__FUNCTION__)))
+	// load from data table
+	const UDataTable* ProgressionDataTable = UPSDataAsset::Get().GetProgressionDataTable();
+	if (!ensureMsgf(ProgressionDataTable, TEXT("ASSERT: [%i] %s:\n'ProgressionDataTable' is not valid!"), __LINE__, *FString(__FUNCTION__)))
 	{
 		return;
 	}
-	SaveGameDataInternal->UnlockAllLevels();
-	SaveDataAsync();
+	UMyDataTable::GetRows(*ProgressionDataTable, ProgressionSettingsDataInternal);
+	
+	checkf(SaveGameDataInternal, TEXT("ERROR: [%i] %hs:\n'SaveGameDataInternal' is null!"), __LINE__, __FUNCTION__);
```

Comment:
Please move `checkf(SaveGameDataInternal)` to be right after `SaveGameDataInternal =` assigning

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/40/commits/c536378208435624a0403f0940005fa786993583